testsuite/tests/singlecluster/observability/conftest.py

```diff
@@ -0,0 +1,75 @@
+"""Conftest for observability tests"""
+
+import time
+
+import backoff
+import pytest
+from openshift_client import selector
+
+from testsuite.kubernetes.monitoring.pod_monitor import PodMonitor
+from testsuite.kubernetes.monitoring.service_monitor import ServiceMonitor
+
+
+@pytest.fixture(scope="module")
+def commit():
+    """Override the commit fixture to do nothing"""
+    return None
+
+
+@pytest.fixture(scope="module", autouse=True)
+def no_observability_metrics_before_enable(prometheus):
+    """Verify no Kuadrant-related observability targets are active before enabling observability"""
+    for _ in range(12):
+        targets = prometheus.get_active_targets()
+        if all("kuadrant" not in t.get("labels", {}).get("job", "") for t in targets):
+            break
+        time.sleep(5)
+    else:
+        raise AssertionError("Observability targets still present in Prometheus before enabling observability")
+
+
+@pytest.fixture(scope="module")
+def observability_setup(gateway, kuadrant, request, wait_for_monitors):  # pylint: disable=unused-argument
+    """Enable observability before the test and wait to retrieve expected monitors"""
+    kuadrant.set_observability(True)
+
+    service_monitors, pod_monitors = wait_for_monitors(present=True)
+    assert service_monitors and pod_monitors, "No monitors appeared after enabling observability"
+
+    def _reset():
+        kuadrant.set_observability(False)
+
+    request.addfinalizer(_reset)
+    return service_monitors, pod_monitors
+
+
+@pytest.fixture(scope="module")
+def wait_for_monitors(cluster, testconfig):
+    """Wait for ServiceMonitor and PodMonitor resources"""
+
+    def _wait(present=True):
```

Comment:
Probably a leftover, but it seems like you using this function only once, you don't need nested fixture with parameter for this.